### ISSUE TITLE
Getting BUFFER_DATA_SIZE for unsized arrays

### ISSUE THREAD

**Issue by tsuoranta** [KhronosGroup/glslang#1852]
I had to make the following modification in order to be able to get buffer data size reflection for unsized arrays:

```
--- a/glslang/MachineIndependent/linkValidate.cpp
+++ b/glslang/MachineIndependent/linkValidate.cpp
@@ -1527,7 +1527,7 @@ int TIntermediate::getBaseAlignment(const TType& type, int& size, int& stride, T
         RoundToPow2(size, alignment);
         stride = size;  // uses full matrix size for stride of an array of matrices (not quite what rule 6/8, but what's expected)
                         // uses the assumption for rule 10 in the comment above
-        size = stride * type.getOuterArraySize();
+        size = type.isUnsizedArray() ? stride : stride * type.getOuterArraySize();
         return alignment;
     }
```
For testing, I used https://github.com/ARM-software/opengl-es-sdk-for-android/blob/master/samples/advanced_samples/ComputeParticles/assets/spawn.cs

Test command:
```
glslangValidator --reflect-strict-array-suffix --reflect-basic-array-suffix --reflect-intermediate-io --reflect-separate-buffers --reflect-all-block-variables --reflect-unwrap-io-blocks -l -q -Od -g -S comp ~/git/opengl-es-sdk-for-android/samples/advanced_samples/ComputeParticles/assets/spawn.cs
```

Output without the modification:
```
Buffer block reflection:
SpawnBuffer: offset -1, type ffffffff, size 0, index -1, binding 0, stages 32, numMembers 1
```

I did not see other way to get data size. Is this modification correct, or is there a better way?

**Comment by johnkslang** [comment]
On the surface, does it change an obviously wrong result to a less obviously wrong result?

That is, it would seem ill-defined to ask the size of something that is not yet sized.  Should there be something more specific about what is being asked that is well-defined?

**Comment by tsuoranta** [comment]
Right, so yes, I do agree this is borderline useless. The only use for this is to fulfill what OpenGL specification requires. Would that count for enough justification?

**Comment by arcady-lunarg** [comment]
This was fixed as part of PR #2231.
